src/utils/sformat.fs

```diff
@@ -1194,7 +1194,7 @@ namespace Microsoft.FSharp.Text.StructuredFormat
             | :? bool   as b -> (if b then "true" else "false")
             | :? char   as c -> "\'" + formatChar true c + "\'"
             | _ -> try  let text = obj.ToString()
-                        text
+                        if text = null then "" else text
```

Comment:
No we do not care about FSLint, but I remember reading isNull and pattern
matching against null makes perf difference in F#. Not sure if this is the
case in proto,  but the only reason fslint is suggesting pattern matching
and isNull is because of perf.
On Oct 29, 2015 10:42 AM, "Robert Jeppesen" <notifications@github.com>
wrote:

> In src/utils/sformat.fs
> <https://github.com/Microsoft/visualfsharp/pull/707#discussion_r43365129>:
>
> > @@ -1194,7 +1194,7 @@ namespace Microsoft.FSharp.Text.StructuredFormat
> >              | :? bool   as b -> (if b then "true" else "false")
> >              | :? char   as c -> "\'" + formatChar true c + "\'"
> >              | _ -> try  let text = obj.ToString()
> > -                        text
> > +                        if text = null then "" else text
>
> isNull is not available in proto, and pattern matching here seems ugly. :)
> Surely we don't have to do everything FSLint says? FSLint would probably
> find tons of issues in this code base.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Microsoft/visualfsharp/pull/707/files#r43365129>.
>